### PR TITLE
Buffer price to Pilot

### PR DESCRIPTION
**Pull request by macano**
Only the last commit is part of this PR.
Merge after https://github.com/Zenmo/Local4Local/pull/32.

### CODE REVIEW

File: webtool/frontend/src/components/configure.tsx [hunk 1]
```diff
@@ -44,6 +48,10 @@ export const Configure: FunctionComponent<{ pilot: Pilot, setPilot: (pilot: Pilo
         setPilot(pilot.withHeatStorage(heatStorage))
     }
 
+    const saveBufferPrice = (bufferPrice: String) => {
+        setPilot(pilot.withBufferPrice(parseFloat(bufferPrice as string) * 0.01))
```
Comment (Erikvv) [inline webtool/frontend/src/components/configure.tsx:52]: Dont need to multiply if input and property is both euros.

File: webtool/gradle/common/src/commonMain/kotlin/Pilot.kt [hunk 2]
```diff
@@ -17,6 +17,7 @@ data class Pilot(
     val windFarms: List<WindFarm> = emptyList(),
     val batteries: List<Battery> = emptyList(),
     val heatStorages: List<HeatStorage> = emptyList(),
+    val bufferPrice_EurpkWh: Double,
```
Comment (Erikvv) [inline webtool/gradle/common/src/commonMain/kotlin/Pilot.kt:20]: EurpkWh -> eurpkWh

File: webtool/frontend/src/components/add-dropdown.tsx [hunk 3]
```diff
@@ -38,6 +40,7 @@ export const AddDropdown: FunctionComponent<{
             <DropdownMenu.Item onSelect={addCompany}>
                 Bedrijf
             </DropdownMenu.Item>
+            <DropdownMenu.Item onSelect={addBufferPrice}>Buffer Price</DropdownMenu.Item>
```
Comment (Erikvv) [inline webtool/frontend/src/components/add-dropdown.tsx:43]: Buffer Price -> Bufferprijs

File: webtool/frontend/src/components/buffer-price.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        saveBufferPrice(solarFarm)
+        hide()
+    }
+
+    return (
+        <Card className="form-box">
+            <BufferPriceHeading/>
+            <form onSubmit={onSubmit}>
+                <div className="radix-grid">
+                    <label className="form-label" htmlFor="bufferPrice_EurpkWh">Onderhoudskosten [€/kWh]</label>
```
Comment (Erikvv) [inline webtool/frontend/src/components/buffer-price.tsx:45]: Incorrect label

File: webtool/frontend/src/components/buffer-price.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                    <input className="form-input" type="number" id="bufferPrice_EurpkWh" name="bufferPrice_EurpkWh" defaultValue={1000}/>
```
Comment (Erikvv) [inline webtool/frontend/src/components/buffer-price.tsx:46]: Too high default. I suggest € 0.01 with step € 0.001

File: webtool/frontend/src/components/buffer-price.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+export const BufferPriceForm: FunctionComponent<{
+    saveBufferPrice: (bufferPrice_EurpkWh: string) => void
+    hide: () => void
+}> = ({saveBufferPrice, hide}) => {
+    const onSubmit = (event: FormEvent) => {
+        event.preventDefault()
+        const form = event.target as HTMLFormElement
+        const formData = new FormData(form)
+        const solarFarm = formData.get("bufferPrice_EurpkWh") as string
```
Comment (Erikvv) [inline webtool/frontend/src/components/buffer-price.tsx:35]: copypasta

**Review by Erikvv — CHANGES_REQUESTED** [review]
Verified as working, just some small feedback.

**Comment by macano** [comment]
work continue in https://github.com/Zenmo/Local4Local/pull/37

**Pull request closed without merging**